choco-solver/src/main/java/org/chocosolver/solver/Solver.java

```diff
@@ -873,8 +889,15 @@ public void findOptimalSolution(ResolutionPolicy policy, RealVar objective, doub
         if (!getObjectiveManager().isOptimization()) {
             set(new ObjectiveManager<RealVar, Double>(objective, policy, precision, true));
         }
-        set(new LastSolutionRecorder(new Solution(), true, this));
-        solve(false);
+        ISolutionRecorder oldRecorder = getSolutionRecorder();
```

Comment:
The idea behind the patch is to clean up the search monitors the findOptimal algorithm uses especially since the monitors it plugs alters the behaviour of the solver (i.e., it's monitor will "restore on close").

I modified my pull request to not touch the pre-existing search recorder.